### PR TITLE
cleanup Watcom make files to be cross-compile ready

### PR DESCRIPTION
**Pull request by jmalak**
cleanup Watcom make files to be cross-compile ready
support only Open Watcom version of tools
CROSS variable is not necessary anymore
build Open Watcom host is configured by PATH and WATCOM variable for host system
makefiles configure target system setup, it is independent on host
conditional code is for NIX platform is guarded by `__UNIX__` macro
consolidate watcom.mif and common/watcom.mif make files to one common/watcom.mif
use OW tools feature (star '*' before tool name) to pass long command line through environment variable for DOS
add support for Linux build
add support for DLL build
add support for resource files
use Watcom debugging format only for 16-bit targets otherwise use Dwarf debugging format (default)

### CODE REVIEW

**Comment by jmalak** [comment]
To dash problem, I don't know if it has sense to support the old versions of Watcom tools.
Open Watcom is free and available in multiple versions. I don't think that all parts of PDCurses is compilable by old versions of Watcom tools. Maybe good decision to define what versions are supported. Also it is a problem with testing. I personaly have only Watcom 11.0c (oldest version available for me) that I can not test older version compatibility. If we define compatibility with Watcom 11.x then I can check if all tools are working with dash and simplify Watcom make files.

**Comment by GitMensch** [comment]
Watcom 11.0c is still the old commercial compiler, not OpenWatcom 1.0, is it?

As far as I'm concerned the only compatibility we should reasonable have is what we also can test without issues - and this would be:

* Open Watcom 1.9 --> "legacy", if anyone still uses OW1.2 he can either use an old compiled PDCurses version, upgrade the local tools, or _try_ to build and may or may not succeed 
* Open Watcom 2.x

I _guess_ that cmake cannot cope with the older versions either.

This is similar to OrangeC (which should be able to use the MinGW Makefiles, overriding the tools) - it just isn't reasonable to support older toolsets here.
Note: it is different from keeping the _source_ compatible to older environments.

@jmalak Back to the topic: What would be needed, ideally with as less variables as possible, to have it cross-compile ready and working with OW19+OW20?
@Bill-Gray and @wmcbrine: Would you be OK with _officially_ making the minimal supported version of the Watcom compiler and toolset as **Open Watcom 1.9**?

**Comment by jmalak** [comment]
Yes, Watcom 11.0c was last version of commercial product. Open Watcom 1.0 is significantly different (by example OW 1.0 has Linux support).
dash/slash problem can be abandon by using dash only for option because all Open Watcom versions from begining support it for OW tools used by PDCurses.

**Comment by GitMensch** [comment]
> dash/slash problem can be abandon by using dash only for option because all Open Watcom versions from begining support it for OW tools used by PDCurses.

Then I'd totally say: always use dashes and only claim support for OW, not for any outdated commercial compiler.

**Comment by jmalak** [comment]
Now only dash options are used, works with Watcom 11.0c but this version doesn't have full set of C90 header files that compilation of PDCurses is imposible
only Open Watcom has sense as build platform

**Comment by jmalak** [comment]
CI build doesn't use clasic make files that can not use same technique as standalone make files
CI make files are generated by CMake that it must use definition compatible with CMake
CMake has stuff for OW already that is better to use it then create new one, even if it is possible

File: common/watcom.mif [hunk 1]
```diff
@@ -1,48 +1,68 @@
...
+CFLAGS += -bt=$(TARGET) -m$(MODEL) -wx -zq -i=$(PDCURSES_SRCDIR)
+!ifeq DEBUG Y
+CFLAGS += -d2 -DDEBUG -DPDCDEBUG
+!else
+CFLAGS += -s -oneatx -DNDEBUG
+!endif
+!ifdef CHTYPE_32
+CFLAGS += -DCHTYPE_32
+!endif
+!ifeq TARGET dos
```
Comment (GitMensch) [inline common/watcom.mif:37]: Shouldn't the WIDE definition moved here, too?
... that's the last issue/question I have with this PR ;-)

Comment (jmalak) [reply common/watcom.mif:37]: I am sorry I still don't finished with PR
I think there are multiple macros which should be added
PDC_WIDE
PDC_FORCE_UTF8
PDC_DLL_BUILD
PDC_RGB

Comment (GitMensch) [reply common/watcom.mif:37]: No problem, just drop a note when you're finished, or even better do this with [re-]questing a review.

Comment (jmalak) [reply common/watcom.mif:37]: I will do.

**Comment by jmalak** [comment]
I finished with all important changes for build make files (manual build)
- cross-compilation 
     you can build on any OW supported host
     you don't need reconfigure for target, it is done by make files
- linux support
     you can build for Linux from any host
- DLL + resource compiler support
     you can create DLL version for 32-bit OS/2 and Windows

**Comment by jmalak** [comment]
It has no impact to Travis CI and Appveyor build
It must be tested manualy by user

**Comment by GitMensch** [comment]
I see. I don't have half of the environments up but assume they will work.
... so PR in full trust.

**Pull request merged**